src/uct/cuda/base/cuda_iface.h

```diff
@@ -50,6 +50,24 @@
     })
 
 
+#define UCT_CUDADRV_CTX_ACTIVE(_state)                             \
+    do {                                                           \
+        CUcontext cur_ctx;                                         \
+        CUdevice dev;                                              \
+        unsigned flags;                                            \
+                                                                   \
+        _state = 0;                                                \
+        /* avoid active state check if no cuda activity */         \
+        if ((CUDA_SUCCESS == cuCtxGetCurrent(&cur_ctx))            \
+            && (NULL != cur_ctx)) {                                \
+            UCT_CUDADRV_FUNC(cuCtxGetDevice(&dev));                \
+            UCT_CUDADRV_FUNC(cuDevicePrimaryCtxGetState(dev,       \
+                                                        &flags,    \
+                                                        &_state)); \
+        }                                                          \
+    } while(0);
```

Comment:
space after while, also you may use just `{}`. According to our coding style,  using of `do {}while(0)` is not needed